src/components/drop-down/drop-down.tsx

```diff
@@ -1,67 +1,57 @@
+import * as keycode from 'keycode';
 import * as React from 'react';
-import {SBComponent, SBStateless} from 'stylable-react-component';
+import {SBComponent} from 'stylable-react-component';
 import {root} from 'wix-react-tools';
+import {Popup} from '../../../src';
 import {ChangeEvent} from '../../types/events';
 import {FormInputProps} from '../../types/forms';
-import {SelectionList, SelectionListItemValue} from '../selection-list';
+import {noop} from '../../utils/noop';
+import {OptionList, SelectionList} from '../selection-list';
 import {CaretDown} from './drop-down-icons';
 import style from './drop-down.st.css';
 
-export interface DropDownInputProps {
-    selectedItem?: DropDownItem;
-    onClick?: React.EventHandler<React.MouseEvent<HTMLDivElement>>;
-}
-
-export const DropDownInput: React.SFC<DropDownInputProps> = SBStateless(props => {
-    return (
-        <div data-automation-id="DROP_DOWN_INPUT" onClick={props.onClick} className="drop-down-input">
-            <span className="label">{props.selectedItem ? props.selectedItem.label : 'Default Text'}</span>
-            <CaretDown className="caret" />
-        </div>
-    );
-}, style);
-
-export interface DropDownListProps extends FormInputProps<SelectionListItemValue> {
-    open: boolean;
-    items?: DropDownItem[];
-    selectedItem?: DropDownItem;
-}
-
-export const DropDownList: React.SFC<DropDownListProps> = SBStateless(props => {
-    if (!props.open) { return null; }
+const KeyCodes: any = {
+    ENTER: keycode('enter'),
+    UP: keycode('up'),
+    DOWN: keycode('down'),
+    SPACE: keycode('space'),
+    ESC: keycode('escape')
+};
 
-    return (
-        <div data-automation-id="DROP_DOWN_LIST">
-            <SelectionList
-                className="drop-down-list"
-                dataSource={props.items!.map((item: DropDownItem) => item.label)}
-                value={props.selectedItem && props.selectedItem.label}
-                onChange={props.onChange}
-            />
-        </div>
-    );
-}, style);
-
-export interface DropDownItem {
-    label: string;
+export interface DropDownProps extends OptionList, FormInputProps<string> {
+    open?: boolean;
+    disabled?: boolean;
+    openOnFocus?: boolean;
+    children?: React.ReactNode;
+    toggleIcon?: React.SFC;
```

Comment:
No need to limit it just to stateless components. Its should be `React.ComponentType<>`